sieve_of_eratosthenes/sieveOfEratosthenes.c

```diff
@@ -0,0 +1,34 @@
+#include <stdio.h>
+#include <string.h>
+#include <math.h>     // for using square root 
+#include <stdbool.h>  // to include bool datatype 
+
+void sieveOfEratosthenes(int n) {
+	bool primes[n+1];
+	int i, sqrtOfn = sqrt(n)+1, j;
+	
+	memset(primes, true, sizeof(primes));
+	primes[0] = false;
+	primes[1] = false;
+
+	for (i = 2; i < sqrtOfn; i++) {
+		if (primes[i]) {
+			for (j = 2*i; j < n+1; j += i) {
+				primes[j] = false;
+			}
+		}
+	}
+
+	for (j = 2; j < n; j++) {
+		if (primes[j]) {
+			printf("%d ", j);
```

Comment:
@singhpratyush I tried declaring array in main and memset in function but it doesnt seem to work properly.
Like, if I input 10 it gives output only 2,3,5. and for 100 also some primes are missing and some non-primes keep getting printed